### DISCUSSION TITLE
unable to hide download button

### DISCUSSION THREAD

**Discussion by madfortech** [tailwindlabs/tailwindcss#12587] [Help] [answered]
![Capture](https://github.com/tailwindlabs/tailwindcss/assets/21261716/96e3cf71-f26c-4ada-aa26-19968f553f69)
![Capture-1](https://github.com/tailwindlabs/tailwindcss/assets/21261716/8045a1d6-b38f-4e54-9c1a-c88e1379cae6)
iframe tag not working javascript code
html video tag working any one help me 

javascript code credit :-
https://stackoverflow.com/questions/57612396/js-onload-controllist-nodownload-for-multiple-videos-in-chrome-only-working

**Comment by wongjn** [comment]
This doesn't seem related to Tailwind – consider seeking assistance elsewhere more related to the technologies directly involved in your problem.

**Comment by alex-krasikau** [comment, marked as answer]
Hey @madfortech!

You're trying to iterate over `videoElem` as if it's an array, but if you're using `getElementById`, this will return a single element, not a collection of elements.

Here is a revised version of the JavaScript that could work if you have multiple videos:
```js
function noDownload() {
  var videoElems = document.querySelectorAll('#notallowed');
  videoElems.forEach(function(video) {
    video.setAttribute('controlsList', 'nodownload');
  });
}

window.onload = noDownload;
```

For the `<iframe>`, the situation is more complex. The `controlsList="nodownload"` attribute works on the `<video>` tag, but not on an `<iframe>`. If the content inside the `<iframe>` is from another domain, you might not have control over it due to cross-domain security policies. If the video is hosted on a domain you control, you should apply the `controlsList` attribute directly to the `<video>` tags on those pages, not through the `<iframe>`.

Hope that helps.

**Reply by madfortech** [reply]
I will use video element.. not working iframe
